frontend/src/app/core/test-utils/MockRouter.ts

```diff
@@ -15,3 +15,20 @@ export class MockRouter extends Router {
     };
   }
 }
+
+export const setUpRouterState = (url: string, router: Router) => {
+  const mockActivatedRouteSnapshot: Partial<ActivatedRouteSnapshot> = {
+    url: [],
+  };
+
+  const mockRouterStateSnapshot: Partial<RouterStateSnapshot> = {
+    url,
+    root: mockActivatedRouteSnapshot as ActivatedRouteSnapshot,
+  };
+
+  const mockRouterState: Partial<RouterState> = {
+    snapshot: mockRouterStateSnapshot as RouterStateSnapshot,
+  };
+
```

Comment:
maybe we can condense this part a bit, so that it's easier to see how the mockRouterState would look like
```suggestion
  const mockRouterState: Partial<RouterState> = {
    snapshot: { url, root: { url: [] } } as RouterStateSnapshot,
  };
```